src/sqlancer/clickhouse/ClickHouseProvider.java

```diff
@@ -102,9 +104,19 @@ public void generateDatabase(ClickHouseGlobalState globalState) throws Exception
 
     @Override
     public SQLConnection createDatabase(ClickHouseGlobalState globalState) throws SQLException {
+        host = globalState.getOptions().getHost();
+        port = globalState.getOptions().getPort();
+        if ("sqlancer".equals(host)) {
```

Comment:
This process actually involves two points:
(1) The user does not specify the host and port, then we need to replace it with the default value, which is different in different databases. This may be better set in xxOptions.java of each database.
(2) If the user sets one of host and port, we should use the value provided by the user to replace the default value of these systems. This should probably be set in xxProvider.java.

Reply:
Yes, there are multiple ways to implement this. It would be great to have this, so that the user knows about the default values.